File: python/isbn-verifier/isbn-verifier.py
import re
def is_valid(isbn):
    isbn=re.findall(r'[0-9X]',isbn)
    a=10; r=0
    if len(isbn)!=10:
        return False
    for i in isbn:
        if i=='X':
            r+=10
        else:
            r+=a*int(i)
        a-=1
    return r%11==0

File: csharp/isbn-verifier/IsbnVerifier.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;
public static class IsbnVerifier{
    public static bool IsValid(string num){
        num = num.Replace("-", "");
        return Regex.IsMatch(num, @"^\d{9}[\dX]$") && num.Reverse().Select((x, u) => (x == 'X' ? 10 : x - '0') * (u + 1)).Sum() % 11 == 0;
    }   
}